packages/docusaurus/src/client/serverEntry.js

```diff
@@ -61,8 +61,11 @@ export default async function render(locals) {
   const scripts = (bundles.js || []).map((b) => b.file);
   const {baseUrl} = locals;
 
-  const renderedHtml = ejs.render(
-    ssrTemplate.trim(),
+  const renderedHtml = eta.render(
+    ssrTemplate
```

Comment:
@nebrelbug hmm, I actually meant that you could add a similar _option_ to Eta code, as is done in ejs. I think it should be in core of render lib. It will also allow testing this feature. WDYT?

Reply:
@lex111 that sounds reasonable to me, in order to improve EJS compatibility. In general I think EJS' solution is a little bit hacky (it will remove whitespace in pre tags, for example) and whitespace removal would ideally be handled by HTML minifiers.

I do think it would be a useful feature to add for those migrating from EJS though, so I think I'll add it. I'll get back to you once I release :smiley:

Reply:
I understand you, but for some users this option may be useful. For example, we do not use pre tags for code blocks, therefore it is preferable for us to remove whitespaces.

Reply:
Sounds good. I just added the feature to Eta, published, and updated this commit with the new version :tada: